src/platform/packages/shared/kbn-lens-embeddable-utils/config_builder/utils.ts

```diff
@@ -135,6 +135,7 @@ export async function getDataView(
     dataView = await dataViewsAPI.get(index, false);
   } catch {
     dataView = await dataViewsAPI.create({
+      id: index,
```

Comment:
without setting an `id`, `dataViewsAPI.get(index, false)` will never get the cached data view